statefulset.go

```diff
@@ -70,16 +67,15 @@ func (c *StatefulSetConfig) CheckAndSetDefaults() error {
 // adds various operations, like delete, status check and update
 type StatefulSetControl struct {
 	StatefulSetConfig
-	statefulSet appsv1.StatefulSet
 	*log.Entry
 }
 
 // Upsert creates or updates a statefulset resource
 func (c *StatefulSetControl) Upsert(ctx context.Context) error {
 	c.Infof("Upsert %v", formatMeta(c.StatefulSet.ObjectMeta))
 
-	collection := c.Client.AppsV1().StatefulSets(c.statefulSet.Namespace)
-	currentResource, err := collection.Get(c.statefulSet.Name, metav1.GetOptions{})
+	collection := c.Client.AppsV1().StatefulSets(c.StatefulSetConfig.StatefulSet.Namespace)
```

Comment:
I think `StatefulSetConfig` gets in the way - can you embed this attribute so the access path would be:
```go
config.StatefulSet.XXX
```
?

There's also an outdated comment on `StatefulSetConfig` that says the resource will only be used if set while, in fact, the resource is mandatory:
```go
// StatefulSetConfig is a StatefulSet control configuration
type StatefulSetConfig struct {
	// StatefulSet is already parsed statefulset, will be used if present   <- HERE
	StatefulSet *appsv1.StatefulSet```